internal/api/api.go

```diff
@@ -25,40 +26,48 @@ type snykApiClient struct {
 }
 
 func (a *snykApiClient) GetOrgIdFromSlug(slugName string) (string, error) {
-	url := a.url + "/v1/orgs"
+	url := GetOrgsApiURL(a.url, slugName)
+
 	res, err := a.client.Get(url)
 	if err != nil {
 		return "", err
 	}
-	defer res.Body.Close()
+	defer func() { _ = res.Body.Close() }()
 
 	body, err := io.ReadAll(res.Body)
 	if err != nil {
 		return "", err
 	}
 
-	var userOrgInfo contract.OrganizationsResponse
-	err = json.Unmarshal(body, &userOrgInfo)
+	var response contract.OrganizationsResponse
+	err = json.Unmarshal(body, &response)
 	if err != nil {
 		return "", err
 	}
 
-	for _, org := range userOrgInfo.Organizations {
-		if org.Slug == slugName {
-			return org.ID, nil
+	organizations := response.Organizations
+	for _, organization := range organizations {
+		if organization.Attributes.Slug == slugName {
+			return organization.Id, nil
 		}
 	}
-
 	return "", fmt.Errorf("org ID not found for slug %v", slugName)
 }
 
+func GetOrgsApiURL(baseURL, slugName string) string {
+	apiVersion := "version=2024-03-12"
+	escapedSlug := url2.PathEscape(slugName)
+	url := fmt.Sprintf("%s/rest/orgs?%s&slug=%s", baseURL, apiVersion, escapedSlug)
+	return url
+}
+
 func (a *snykApiClient) GetDefaultOrgId() (string, error) {
 	url := a.url + "/rest/self?version=" + constants.SNYK_API_VERSION
 	res, err := a.client.Get(url)
 	if err != nil {
 		return "", fmt.Errorf("unable to retrieve org ID: %w", err)
 	}
-	defer res.Body.Close()
+	defer func() { _ = res.Body.Close() }()
```

Comment:
I don't understand the value this adds. The original defer was fine (ignoring a close error is reasonable).

Do we need to adjust our linter?

Reply:
Or deactivate the IntelliJ linting rule.